tests/gtest/avifgridapitest.cc

```diff
@@ -15,139 +14,233 @@ using testing::ValuesIn;
-                                 ValuesIn(kBitDepths), ValuesIn(kPixelFormats),
-                                 /*create_alpha=*/Values(false, true),
-                                 /*expected_success=*/Values(false)));
-
-// Special case depending on the cell count and the chroma subsampling.
-INSTANTIATE_TEST_SUITE_P(ValidOddHeight, GridApiTest,
-                         Combine(/*horizontal=*/Values(Cell{1, 64}),
-                                 /*vertical=*/Values(Cell{1, 65}, Cell{2, 65}),
-                                 ValuesIn(kBitDepths),
-                                 Values(AVIF_PIXEL_FORMAT_YUV444,
-                                        AVIF_PIXEL_FORMAT_YUV422,
-                                        AVIF_PIXEL_FORMAT_YUV400),
-                                 /*create_alpha=*/Values(false, true),
-                                 /*expected_success=*/Values(true)));
-INSTANTIATE_TEST_SUITE_P(InvalidOddHeight, GridApiTest,
-                         Combine(/*horizontal=*/Values(Cell{1, 64}),
-                                 /*vertical=*/Values(Cell{2, 65}),
-                                 ValuesIn(kBitDepths),
-                                 Values(AVIF_PIXEL_FORMAT_YUV420),
-                                 /*create_alpha=*/Values(false, true),
-                                 /*expected_success=*/Values(false)));
-
-// Special case depending on the cell count and the cell size.
-INSTANTIATE_TEST_SUITE_P(ValidOddDimensions, GridApiTest,
-                         Combine(/*horizontal=*/Values(Cell{1, 1}),
-                                 /*vertical=*/Values(Cell{1, 65}),
-                                 ValuesIn(kBitDepths), ValuesIn(kPixelFormats),
-                                 /*create_alpha=*/Values(false, true),
-                                 /*expected_success=*/Values(true)));
-INSTANTIATE_TEST_SUITE_P(InvalidOddDimensions, GridApiTest,
-                         Combine(/*horizontal=*/Values(Cell{2, 1}),
-                                 /*vertical=*/Values(Cell{1, 65}, Cell{2, 65}),
-                                 ValuesIn(kBitDepths), ValuesIn(kPixelFormats),
-                                 /*create_alpha=*/Values(false, true),
-                                 /*expected_success=*/Values(false)));
+TEST(GridApiTest, CellsOfDifferentDimensions) {
+  for (avifPixelFormat pixel_format :
+       {AVIF_PIXEL_FORMAT_YUV444, AVIF_PIXEL_FORMAT_YUV422,
+        AVIF_PIXEL_FORMAT_YUV420, AVIF_PIXEL_FORMAT_YUV400}) {
+    // Right-most cells are narrower.
+    EXPECT_EQ(
+        EncodeDecodeGrid({{{100, 100}, {100, 100}, {66, 100}}}, pixel_format),
+        AVIF_RESULT_OK);
+    // Bottom-most cells are smaller.
+    EXPECT_EQ(EncodeDecodeGrid({{{100, 100}, {100, 100}},
+                                {{100, 100}, {100, 100}},
+                                {{100, 66}, {100, 66}}},
+                               pixel_format),
+              AVIF_RESULT_OK)
+        << pixel_format;
```

Comment:
Nit: Delete `<< pixel_format`? Or add it to the other `EXPECT_EQ` statements for consistency?